### PR TITLE
Add system check encouraging Smarty3

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Add system check encouraging Smarty3

Before
----------------------------------------
As of 5.69 Smarty3 works with core & people should start using it in preparation for us doing a hard swap. However, there is nothing to warn site users of that

After
----------------------------------------
image

Technical Details
----------------------------------------
We probably want a blog or docs update to point to for more information. In addition it would be good to check the civix version of installed extensions & specify whether they have been updated for Smarty3

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by seamuslee001** [comment]
My immediate thought is that Warning might be too high and maybe a Notice is better and also I feel like for your average staff member this may not really mean a heck of a lot so we need to probably nuance the language a bit IMO

**Comment by eileenmcnaughton** [comment]
@seamuslee001 yeah maybe - I kinda wanted to get it in a bit higher & then tone it down cos that way the developers currently looking at master would get the serious warning & hopefully update their dev & sort out their extensions so they can use it

If you don't do 'warning' there is no pop up - which means that people won't see - & if we want to consider hard-switching before the next ESR then we really need people to process the fact that they might need to think about it

File: CRM/Utils/Check/Component/Smarty.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  public function checkSmarty3(): array {
+    $messages = [];
+    if (!defined('CIVICRM_SMARTY3_AUTOLOAD_PATH')) {
+      $smartyPath = rtrim(\Civi::paths()->getPath('[civicrm.packages]/'), '/') . DIRECTORY_SEPARATOR . 'smarty3/vendor/autoload.php';
+      $messages[] = new CRM_Utils_Check_Message(
+        __FUNCTION__,
+        ts('We are in the process of migrating from Smarty2 onto Smarty3 and then Smarty4 for performance and security reasons. '
+          . 'As of CiviCRM 5.69 switching to Smarty3 is optional but recommended.'
+          . ' In order to switch you need to add the following to your civicrm.settings.php file: ')
+        . "<pre>define('CIVICRM_SMARTY3_AUTOLOAD_PATH', $smartyPath');</pre>"
```
Comment (colemanw) [inline CRM/Utils/Check/Component/Smarty.php:35]: There is an opening single-quote missing before $smartyPath
```suggestion
        . "<pre>define('CIVICRM_SMARTY3_AUTOLOAD_PATH', '$smartyPath');</pre>"
```

**Comment by eileenmcnaughton** [comment]
That reads better - thanks @totten

**Comment by totten** [comment]
Test failure is probably unrelated. Just in case, I've requested a narrow retest: https://test.civicrm.org/job/CiviCRM-Manual-Test/112/

**Comment by eileenmcnaughton** [comment]
re-test passed

**Pull request merged**